plugins/touchegg/Main.vala

```diff
@@ -0,0 +1,103 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+public class Gala.Plugins.Touchegg.Plugin : Gala.Plugin {
+    private Gala.WindowManager? wm = null;
+    private Client? client = null;
+
+    /**
+     * Percentage of the animation to be completed to apply the action.
+     */
+    private const int SUCCEESS_THRESHOLD = 20;
+
+    public override void initialize (Gala.WindowManager wm) {
+        this.wm = wm;
+
+        this.client = new Client ();
+        this.client.on_gesture_begin.connect ((gesture) => Idle.add (() => {
+            this.on_handle_gesture (gesture, "begin");
+            return false;
+        }));
+        this.client.on_gesture_update.connect ((gesture) => Idle.add (() => {
+            this.on_handle_gesture (gesture, "update");
+            return false;
+        }));
+        this.client.on_gesture_end.connect ((gesture) => Idle.add (() => {
+            this.on_handle_gesture (gesture, "end");
+            return false;
+        }));
+        this.client.run ();
+    }
+
+    public override void destroy () {
+        // TODO (José Expósito) This is not being called, but I should close the socket here
+    }
+
+
+    private void on_handle_gesture (Gesture gesture, string event) {
+        // debug (@"Gesture $(event): $(gesture.type) - $(gesture.direction) - $(gesture.fingers) fingers - $(gesture.percentage)% - $(gesture.elapsed_time) - $(gesture.performed_on_device_type)");
+        var hints = this.build_hints_from_gesture (gesture, event);
+
+        if (this.is_open_workspace_gesture (gesture)) {
+            this.wm.workspace_view.open (hints);
+        }
+
+        if (this.is_close_workspace_gesture (gesture)) {
+            this.wm.workspace_view.close (hints);
+        }
+    }
+
+    private GLib.HashTable<string, Variant> build_hints_from_gesture (Gesture gesture, string event) {
+        var hints = new GLib.HashTable<string, Variant> (str_hash, str_equal);
+        hints.insert ("manual_animation", new Variant.boolean (true));
+        hints.insert ("event", new Variant.string (event));
+        hints.insert ("percentage", new Variant.int32 (gesture.percentage));
+
+        if (event == "end") {
+            hints.insert ("cancel_action", new Variant.boolean (gesture.percentage < SUCCEESS_THRESHOLD));
+        }
+
+        return hints;
+    }
+
+    // TODO (José Expósito) It'll be nice to be able to configure this from Switchboard
+    // https://github.com/elementary/switchboard-plug-mouse-touchpad/issues/16
+
+    private bool is_open_workspace_gesture (Gesture gesture) {
```

Comment:
This simply can be a `static` method or part of the `Gesture` class.

Reply:
This solution is temporary, as mentioned in the `TODO` I'll add configuration options for this and I'll move this 2 methods to a different class.